### ISSUE TITLE
Result pagination

### ISSUE THREAD

**Issue by rkistner** [akirsman/Snapshots#2]
The current script only lists the first 10000 changed pages for each snapshot, so gives inaccurate results for any volume larger than around 5GB.

I changed the relevant part to this to use pagination:

```py
        changed = 0
        token = None
        while True:
            if token:
                page = ebs.list_changed_blocks(FirstSnapshotId = sid_prev,SecondSnapshotId = sid, NextToken=token, MaxResults=10000)
            else:
                page = ebs.list_changed_blocks(FirstSnapshotId = sid_prev,SecondSnapshotId = sid, MaxResults=10000)
            
            changed += len(page['ChangedBlocks'])
            if 'NextToken' not in page:
                break
            token = page['NextToken']
            if token is None:
                break
        print(date + ',' + time + ',' + v + "," + sid_prev + "," + sid + "," + str(changed * blockSize))
```

There is probably a similar issue in listing the snapshots, but it's less likely to run into the limit there.

**Comment by GauravES** [comment]
`The current script only lists the first 10000 changed pages for each snapshot, so gives inaccurate results for any volume larger than around 5GB.`

I presume you meant 10000 changed blocks and inaccurate results for any snapshot with changes exceeding ~5GB from the last snapshot.

Were you able to finally reconcile your script generated numbers with AWS billing?

**Comment by rkistner** [comment]
> I presume you meant 10000 changed blocks and inaccurate results for any snapshot with changes exceeding ~5GB from the last snapshot.

Yes, likely that. I think I just went with the simple "works for volume < 5GB". 

I only used this to get estimates when using a new snapshot schedule. I never properly checked whether it matches the billing numbers, but it was close enough for my estimates.
